vg/font_syscall.go

```diff
@@ -7,31 +7,28 @@
 package vg
 
 import (
-	"go/build"
 	"os"
 	"path/filepath"
-)
 
-const (
-	// importString is the import string expected for
-	// this package.  It is used to find the font
-	// directory included with the package source.
-	importString = "gonum.org/v1/plot/vg"
+	"golang.org/x/tools/go/packages"
 )
 
 func init() {
 	FontDirs = initFontDirs()
 }
 
-// InitFontDirs returns the initial value for the FontDirectories variable.
+// initFontDirs returns the initial value for the FontDirs variable.
 func initFontDirs() []string {
+	const fonts = "gonum.org/v1/plot/vg/fonts"
+
 	dirs := filepath.SplitList(os.Getenv("VGFONTPATH"))
 
-	if pkg, err := build.Import(importString, "", build.FindOnly); err == nil {
-		p := filepath.Join(pkg.Dir, "fonts")
-		if _, err := os.Stat(p); err == nil {
-			dirs = append(dirs, p)
-		}
+	cfg := &packages.Config{
+		Mode: packages.NeedFiles,
+	}
+	pkgs, err := packages.Load(cfg, fonts)
```

Comment:
@sbinet  after messing around a bit I think I understand the issue I was seeing a bit better. This call makes the underlying build tool load the package, which results in a `go.mod` file if called within another module.

I was actually able to get this to work by changing this line to:
```
pkgs, err := packages.Load(cfg, fonts, "mod=readonly")
```

Per the [docs](https://godoc.org/golang.org/x/tools/go/packages):
```
Load passes most patterns directly to the underlying build tool....
```

Reply:
Yes, readonly doesn't mean it won't modify, but that it will fail if it can't modify. See https://github.com/golang/go/issues/26850#issuecomment-411903910